<a href="https://colab.research.google.com/github/perigor/EngenhariadeDados-Growdev/blob/main/Atividade_Analise_dados_Pandas_Semana_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

![Growdev](https://www.growdev.com.br/assets/images/logo_growdev.png)

![Formação Engenharia de Dados](https://d335luupugsy2.cloudfront.net/cms/files/524558/1707226566/$occu5m8t1op)

Este material tem como objetivo incentivar a prática dos conteúdos estudados até o momento, e a pesquisa. Tenha em consideração que o exercício foi pensado para incentivar a prática dos conteúdos estudados até então, porém aprender a pesquisar faz parte do processo de aprendizagem.

Fique tranquilo, faça os exercícios com calma, e boa prática!


#Orientações:

 - Faça uma copia deste arquivo
 - Após concluir os exercicios, suba seu exercicio para o GitHub e compartilhe o link no **google** Classroom



Utilize o dataset contido nesse [link](https://www.kaggle.com/datasets/upadorprofzs/testes), e responda as questões abaixo.

O dataset contém dados estatísticos do IBGE relativos à escolaridade, renda entre outros campos avaliados pela  PNAD (Pesquisa Nacional por Amostra de Domicílios).

Os dados podem representar informações relevantes e reveladoras sobre a distribuição de renda e possivelmente alguns indicadores sobre tais situações encontradas a partir dos dados.

#Exercícios

In [1]:
# Importando biblioteca

import pandas as pd


In [2]:
# dados

df_pnad = pd.read_csv("/content/dados.csv")
df_pnad.head()

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura
0  11     0     23    8              12    800  1.603808
1  11     1     23    2              12   1150  1.739790
2  11     1     35    8              15    880  1.760444
3  11     0     46    2               6   3500  1.783158
4  11     1     47    8               9    150  1.690631

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura  Faixa Etária
0      11     0     23    8              12    800  1.603808  21 a 35 anos
1      11     1     23    2              12   1150  1.739790  21 a 35 anos
2      11     1     35    8              15    880  1.760444  36 a 50 anos
3      11     0     46    2               6   3500  1.783158  36 a 50 anos
4      11     1     47    8               9    150  1.690631  36 a 50 anos
...    ..   ...    ...  ...             ...    ...       ...           ...
76835  53     1     46    2              11    812  1.687030  36 a 50 anos
76836  53     0     30    4               7   1500  1.792934  21 a 35 anos
76837  53     0     32    8              12   1300  1.830587  21 a 35 anos
76838  53     0     57    8               4   1500  1.726344  51 a 65 anos
76839  53     0     38    8               4    900  1.658305  36 a 50 anos

[76840 rows x 8 columns]

Qual a quantidade de pessoas nas seguintes faixas etárias
- 20 anos ou menos
- 21 a 35 anos
- 36 a 50 anos
- 51 a 65 anos
- maiores de 65 anos



In [3]:

# Definindo as faixas etárias
faixas_etarias = [
    (df_pnad['Idade'] <= 20, '20 anos ou menos'),
    ((df_pnad['Idade'] >= 21) & (df_pnad['Idade'] <= 35), '21 a 35 anos'),
    ((df_pnad['Idade'] >= 36) & (df_pnad['Idade'] <= 50), '36 a 50 anos'),
    ((df_pnad['Idade'] >= 51) & (df_pnad['Idade'] <= 65), '51 a 65 anos'),
    (df_pnad['Idade'] > 65, 'maiores de 65 anos')
]

# Contagem
contagem_faixas = {descricao: len(df_pnad[condicao]) for condicao, descricao in faixas_etarias}

# Prints
for faixa, quantidade in contagem_faixas.items():
    print(f'{faixa}: {quantidade} pessoas')

20 anos ou menos: 764 pessoas
21 a 35 anos: 20654 pessoas
36 a 50 anos: 31770 pessoas
51 a 65 anos: 20006 pessoas
maiores de 65 anos: 3646 pessoas


Qual é a média de escolaridade em cada faixa etária?


In [29]:
# Definir as faixas etárias
bins = [0, 20, 35, 50, 65, 100]  #Intervalo com bas na faixa etária
labels = ['0 a 20 anos', '21 a 35 anos', '36 a 50 anos', '51 a 65 anos', 'maiores de 65 anos'] #Fornece rótulos das faixas etárias

# Nova coluna para as faixas etárias
df_pnad['Faixa Etária'] = pd.cut(df_pnad['Idade'], bins=bins, labels=labels, right=True)

# Média de anos de estudo em cada faixa etária
media_escolaridade_faixa_etaria = df_pnad.groupby('Faixa Etária')['Anos de Estudo'].mean()

# Print
media_escolaridade_faixa_etaria.reset_index()

Faixa Etária  Anos de Estudo
0         0 a 20 anos        9.695026
1        21 a 35 anos       10.813111
2        36 a 50 anos        9.764274
3        51 a 65 anos        8.267320
4  maiores de 65 anos        5.842293

Qual a UF que concentra a maior renda?


In [22]:
# Localizando a linha com valor máximo dentro de RENDA
indice_max_renda = df_pnad['Renda'].idxmax()

# Localizando  a UF associada ao valor máximo
uf_maior_renda = df_pnad.loc[indice_max_renda, 'UF']
renda_maxima = df_pnad.loc[indice_max_renda, 'Renda']

print(f"A UF que concentra a maior renda é a Bahia, número {uf_maior_renda} com uma renda de R${renda_maxima:.2f}")



A UF que concentra a maior renda é a Bahia, número 29 com uma renda de R$200000.00


Qual a renda média de pessoas pretas e brancas de homens e mulheres?


Códigos conforme DF


- 0	Masculino
- 1	Feminino

________________________

- 0	Indígena
- 2	branca
- 4	Preta
- 6 Amarela
- 8	Parda
- 9	Sem declaração


In [12]:
# Identificação de cor conforme DF
cor = {0: 'Indígena', 2: 'Branca', 4: 'Preta', 6: 'Amarela', 8: 'Parda', 9: 'Sem declaração'}

# Identificação de sexo conforme DF
sexo = {0: 'Masculino', 1: 'Feminino'}

# Criando 'Cor Descrição' e 'Sexo Descrição' usando os mapeamentos
df_pnad['Cor Descrição'] = df_pnad['Cor'].map(cor)
df_pnad['Sexo Descrição'] = df_pnad['Sexo'].map(sexo)

# Filtrar apenas as pessoas pretas e brancas
df_brancos_pretos = df_pnad[df_pnad['Cor'].isin([2, 4])]

# Agrupando por Sexo Descrição e Cor Descrição e calculando a renda média
renda_media = df_brancos_pretos.groupby(['Sexo Descrição', 'Cor Descrição'])['Renda'].mean()

# Print
renda_media.reset_index()

Sexo Descrição Cor Descrição        Renda
0       Feminino        Branca  2109.866750
1       Feminino         Preta  1134.596400
2      Masculino        Branca  2925.744435
3      Masculino         Preta  1603.861687

Qual a renda média concentrada em cada faixa etária?




In [27]:
# Definir faixas etárias
bins = [0, 20, 35, 50, 65, 100]
labels = ['0 a 20 anos', '21 a 35 anos', '36 a 50 anos', '51 a 65 anos', 'maiores de 65 anos']

# 2. Criando uma nova coluna para as faixas etárias
df_pnad['Faixa Etária'] = pd.cut(df_pnad['Idade'], bins=bins, labels=labels, right=True)

# 3. Agrupar por Faixa Etária e calcular a renda média
renda_media_faixa_etaria = df_pnad.groupby('Faixa Etária')['Renda'].mean()

# Mostrar o resultado
renda_media_faixa_etaria.reset_index()

Faixa Etária        Renda
0         0 a 20 anos   985.348168
1        21 a 35 anos  1748.286143
2        36 a 50 anos  2113.756091
3        51 a 65 anos  2150.196741
4  maiores de 65 anos  1831.231487

Faixa Etária	Renda
0	20 anos ou menos	985.348168
1	21 a 35 anos	1748.286143
2	36 a 50 anos	2113.756091
3	51 a 65 anos	2150.196741
4	maiores de 65 anos	1831.231487

Qual a renda média nas faixas de anos de estudo abaixo?
- 2 anos ou menos
- 3 a 6 anos
- 7 a 10 anos
- 10 a 12 anos
- 13 anos ou mais


Qual a escolaridade entre indígenas, para homens e mulheres?


In [31]:
# Adicionando colunas 'Cor Descrição' e 'Sexo Descrição' usando os mapeamentos
df_pnad['Cor Descrição'] = df_pnad['Cor'].map(cor)
df_pnad['Sexo Descrição'] = df_pnad['Sexo'].map(sexo)

# Filtrando apenas os indígenas
df_indigenas = df_pnad[df_pnad['Cor'] == 0]

# Calculando a escolaridade média por sexo entre os indígenas
escolaridade_media_indigenas = df_indigenas.groupby('Sexo Descrição')['Anos de Estudo'].mean()

# Print
escolaridade_media_indigenas.reset_index()

Sexo Descrição  Anos de Estudo
0       Feminino        9.108911
1      Masculino        7.875000

Qual a média de anos de estudo entre brancos e negros?


In [32]:


# Adicionar coluna 'Cor Descrição' usando o mapeamento
df_pnad['Cor Descrição'] = df_pnad['Cor'].map(cor)

# Filtrando brancos e negros
df_brancos_negros = df_pnad[df_pnad['Cor'].isin([2, 4])]

# Calculando a média de anos de estudo para cada grupo
media_anos_estudo_brancos_negros = df_brancos_negros.groupby('Cor Descrição')['Anos de Estudo'].mean()

# Print
media_anos_estudo_brancos_negros.reset_index()

Cor Descrição  Anos de Estudo
0        Branca       10.490303
1         Preta        8.861518

Qual a média de anos de estudo e de renda para mulheres brancas, e para mulheres negras?

In [33]:
# Filtrando mulheres brancas
mulheres_brancas = df_pnad[(df_pnad['Sexo'] == 1) & (df_pnad['Cor'] == 2)]

# Filtrando  mulheres negras
mulheres_negras = df_pnad[(df_pnad['Sexo'] == 1) & (df_pnad['Cor'] == 4)]

# Calcullando média de anos de estudo e renda para cada grupo
media_mulheres_brancas = mulheres_brancas[['Anos de Estudo', 'Renda']].mean()
media_mulheres_negras = mulheres_negras[['Anos de Estudo', 'Renda']].mean()

# Print
print("Média de anos de estudo e renda para mulheres brancas:")
print(media_mulheres_brancas)

print("\nMédia de anos de estudo e renda para mulheres negras:")
print(media_mulheres_negras)

Média de anos de estudo e renda para mulheres brancas:
Anos de Estudo      11.297578
Renda             2109.866750
dtype: float64

Média de anos de estudo e renda para mulheres negras:
Anos de Estudo       9.522326
Renda             1134.596400
dtype: float64


 Qual é a UF com maior média de escolaridade e qual a UF com maior média de renda?

In [35]:
# Média de escolaridade por UF
media_escolaridade_uf = df_pnad.groupby('UF')['Anos de Estudo'].mean()

# Média de renda por UF
media_renda_uf = df_pnad.groupby('UF')['Renda'].mean()

# 3. Identificando a UF com maior média de escolaridade e maior média de renda
uf_maior_media_escolaridade = media_escolaridade_uf.idxmax()
uf_maior_media_renda = media_renda_uf.idxmax()

# Mostrar os resultados
print(f"UF com maior média de escolaridade: {uf_maior_media_escolaridade} ({media_escolaridade_uf.max():.2f} anos de estudo)")
print(f"UF com maior média de renda: {uf_maior_media_renda} (R$ {media_renda_uf.max():.2f})")

#53	Distrito Federal

UF com maior média de escolaridade: 53 (11.91 anos de estudo)
UF com maior média de renda: 53 (R$ 4241.95)


Elabore você uma pergunta sobre a base de dados e escreva o código para responder a pergunta.

"Qual a média de Altura por UF?"


In [45]:
uf_map = {
    '11': 'Rondônia',
    '12': 'Acre',
    '13': 'Amazonas',
    '14': 'Roraima',
    '15': 'Pará',
    '16': 'Amapá',
    '17': 'Tocantins',
    '21': 'Maranhão',
    '22': 'Piauí',
    '23': 'Ceará',
    '24': 'Rio Grande do Norte',
    '25': 'Paraíba',
    '26': 'Pernambuco',
    '27': 'Alagoas',
    '28': 'Sergipe',
    '29': 'Bahia',
    '31': 'Minas Gerais',
    '32': 'Espírito Santo',
    '33': 'Rio de Janeiro',
    '35': 'São Paulo',
    '41': 'Paraná',
    '42': 'Santa Catarina',
    '43': 'Rio Grande do Sul',
    '50': 'Mato Grosso do Sul',
    '51': 'Mato Grosso',
    '52': 'Goiás',
    '53': 'Distrito Federal'
}

# Convertendo a coluna UF para o tipo string
df_pnad['UF'] = df_pnad['UF'].astype(str)

# Mapeado os códigos de UF para as descrições usando o dicionário uf_map
df_pnad['UF Descrição'] = df_pnad['UF'].map(uf_map)

# Média de altura por UF
media_altura_uf = df_pnad.groupby('UF Descrição')['Altura'].mean()

# Print

media_altura_uf.reset_index()

UF Descrição    Altura
0                  Acre  1.699113
1               Alagoas  1.693907
2                 Amapá  1.697891
3              Amazonas  1.699454
4                 Bahia  1.699282
5                 Ceará  1.700317
6      Distrito Federal  1.700896
7        Espírito Santo  1.701513
8                 Goiás  1.699623
9              Maranhão  1.699708
10          Mato Grosso  1.698339
11   Mato Grosso do Sul  1.700184
12         Minas Gerais  1.699988
13               Paraná  1.699434
14              Paraíba  1.699266
15                 Pará  1.698197
16           Pernambuco  1.698141
17                Piauí  1.700189
18  Rio Grande do Norte  1.699135
19    Rio Grande do Sul  1.698710
20       Rio de Janeiro  1.700365
21             Rondônia  1.703967
22              Roraima  1.698680
23       Santa Catarina  1.698933
24              Sergipe  1.702222
25            São Paulo  1.699321
26            Tocantins  1.699969